### PR TITLE
Use \u escapes in json where possible

### PR DESCRIPTION
**Pull request by aswan**
Don't use \x to escape valid non-printable unicode code points in
json output.

### CODE REVIEW

File: zng/escape.go [hunk 1]
```diff
@@ -28,18 +29,28 @@ func Escape(data []byte) string {
...
+//
+// This function is used to escape non-printable characters with \x syntax.
+// If invalidOnly is false, all non-printable characters are escaped.  If
+// invalidOnly is true, only bytes that cannot be decoded as UTF-8 are escaped.
+func EscapeUTF8(data []byte, invalidOnly bool) string {
 	var out []byte
 	var start int
 	for i, r := range *(*string)(unsafe.Pointer(&data)) {
-		if r < 0x20 || r == 0x7f || r == utf8.RuneError {
+		if !unicode.IsPrint(r) && (!invalidOnly || r == utf8.RuneError) {
```
Comment (nwt) [inline zng/escape.go]: Maybe change the parameter name to validOnly, which would be clearer here than !invalidOnly.

File: tests/suite/format/json.go [hunk 2]
```diff
@@ -21,3 +21,21 @@ const jsonInput = `
...
+const stringIn = `
+#0:record[s:string]
+0:[\x03;]
+`
+
+const stringExpected = `
+{"s":"\u0003"}
+`
+
+var JsonString = test.Internal{
```
Comment (nwt) [inline tests/suite/format/json.go]: s/JsonString/JSONString/

**Comment by aswan** [comment]
I'm going to revise this, but in the mean time if you can put your finger on what's confusing, I would be happy to try to clarify things...

**Comment by aswan** [comment]
Bailing on this to maintain compatibility with Zeek, I'll bring the comment clarification into a forthcoming PR.

**Pull request closed without merging**